perceval/backends/git.py

```diff
@@ -65,7 +65,7 @@ def __init__(self, uri, gitpath, cache=None, origin=None):
         self.gitpath = gitpath
 
     @metadata
-    def fetch(self, from_date=DEFAULT_DATETIME):
+    def fetch(self, from_date=DEFAULT_DATETIME, branches=[]):
```

Comment:
In my opinion, the default value of branches should be `None`. When `None` is given, no filters are applied and all commits are returned. When the list is empty, it means that there's a filter but not branches were given, so no commits will be returned. It could only be useful just for profiling purposes but I think it fits well with the meaning of the filter. In addition, a message can be logged to warns about that fact.